packages/ag-charts-enterprise/src/features/annotations/axisButton.ts

```diff
@@ -82,29 +87,38 @@ export class AxisButton extends BaseModuleInstance implements _ModuleSupport.Mod
         this.toggleVisibility(false);
     }
 
-    private getButtonCoordinates(event: _ModuleSupport.PointerInteractionEvent<'hover' | 'drag'>) {
+    private getButtonCoordinates({ x, y }: Coords) {
         const {
             axisCtx: { direction, position },
             seriesRect,
+            snap,
+            axisCtx,
+            padding,
         } = this;
 
         const { clientWidth: buttonWidth, clientHeight: buttonHeight } = this.button;
 
-        const [minY, maxY] = [seriesRect.y, seriesRect.y + seriesRect.height - buttonHeight];
-        const [minX, maxX] = [seriesRect.x, seriesRect.x + seriesRect.width - buttonWidth];
+        const [minY, maxY] = [seriesRect.y, seriesRect.y + seriesRect.height];
+        const [minX, maxX] = [seriesRect.x, seriesRect.x + seriesRect.width];
+
+        if (snap) {
+            x = convert(invert(x - seriesRect.x, axisCtx), axisCtx) + seriesRect.x;
+            y = convert(invert(y - seriesRect.y, axisCtx), axisCtx) + seriesRect.y;
+        }
```

Comment:
Future idea (not for this PR) - I've had to do this as well, perhaps we could add a `snap(pointInPixels): pointInPixels` method on the axis & axisCtx, as some scales may be able to do this more efficiently than converting back and forth?